src/test/java/org/folio/entitlement/support/extensions/impl/KongGatewayExtension.java

```diff
@@ -33,10 +44,40 @@ public void beforeAll(ExtensionContext extensionContext) {
       CONTAINER.start();
     }
 
-    System.setProperty(KONG_URL_PROPERTY, getUrlForExposedPort(8001));
+    var enableWiremock = isWireMockEnabled(extensionContext);
+
+    String kongUrl = getUrlForExposedPort(8001);
+    if (enableWiremock) {
+      try {
+        var runContainer = WireMockExtension.class.getDeclaredMethod("runContainer");
+        runContainer.setAccessible(true);
+        runContainer.invoke(null);
+        var getUrlForExposedPort = WireMockExtension.class.getDeclaredMethod("getUrlForExposedPort");
+        getUrlForExposedPort.setAccessible(true);
+        String wireMockUrl = getUrlForExposedPort.invoke(null).toString();
+        wireMockClient =
+          new WireMock(new URI(wireMockUrl).getHost(), new URI(wireMockUrl).getPort());
+        kongUrl = wireMockUrl;
+      } catch (Exception e) {
+        throw new RuntimeException("Failed to start Wiremock", e);
+      }
```

Comment:
probably can be moved to dedicated method to make setup method more readable - smth like `prepareAndRunWiremock`